Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -1609,26 +1614,32 @@ private async Task<object> CopyObjectRequestAsync(CopyObjectRequestArgs args, Ca
     /// <returns>bytes read in a byte array</returns>
     internal async Task<byte[]> ReadFullAsync(Stream data, int currentPartSize)
     {
-        var result = new byte[currentPartSize];
         var totalRead = 0;
-        while (totalRead < currentPartSize)
-        {
-            var curData = new byte[currentPartSize - totalRead];
-            var curRead = await data.ReadAsync(curData, 0, currentPartSize - totalRead).ConfigureAwait(false);
-            if (curRead == 0) break;
-            for (var i = 0; i < curRead; i++) result[totalRead + i] = curData[i];
-            totalRead += curRead;
-        }
+        var shared = ArrayPool<byte>.Shared;
+        var curData = shared.Rent(currentPartSize - totalRead);
 
-        if (totalRead == 0) return null;
+        try
+        {
+            while (totalRead != currentPartSize &&
+                   totalRead < currentPartSize)
+            {
+                curData = shared.Rent(currentPartSize - totalRead);
```

Comment:
This is a memory leak. You are not freeing the rented buffer from line 1619.

Reply:
Yes. This part of the code is not finalized yet and might change completely without any `while` if possible.
We just need to make sure there is no regression.